dcicutils/scripts/publish_to_pypi.py

```diff
@@ -0,0 +1,302 @@
+    if package_version.startswith("v"):
+        package_version = package_version[1:]
+    return package_version
+
+
+def get_package_name() -> str:
+    """
+    Returns the base name of the current git repo name.
+    """
+    package_name, _ = execute_command("git config --get remote.origin.url".split(" "))
+    package_name = os.path.basename(package_name[0])
+    if package_name.endswith(".git"):
+        package_name = package_name[:-4]
+    return package_name
+
+
+def get_package_directories() -> list:
+    """
+    Returns a list of directories constituting the Python package of the current repo,
+    according to the pyproject.toml file.
+    """
+    package_directories = []
+    with open("pyproject.toml", "r") as f:
+        pyproject_toml = toml.load(f)
+    pyproject_package_directories = pyproject_toml["tool"]["poetry"]["packages"]
+    for pyproject_package_directory in pyproject_package_directories:
+        package_directory = pyproject_package_directory.get("include")
+        if package_directory:
+            package_directory_from = pyproject_package_directory.get("from")
+            if package_directory_from:
+                package_directory = os.path.join(package_directory_from, package_directory)
+            package_directories.append(package_directory)
+    return package_directories
+
+
+def execute_command(command_argv: Union[list, str], lines_containing: str = None) -> Tuple[list, int]:
+    """
+    Executes the given command as a command-line subprocess, and returns a tuple whose first element
+    is the list of lines from the output of the command, and the second element is the exit status code.
+    """
+    def cleanup_funny_output(output: str) -> str:
+        return output.replace("('", "").replace("',)", "").replace("\\n\\n", "\n").replace("\\n", "\n")
+
+    if isinstance(command_argv, str):
+        command_argv = [arg for arg in command_argv.split(" ") if arg.strip()]
+    if DEBUG:
+        PRINT(f"DEBUG: {' '.join(command_argv)}")
+    result = subprocess.run(command_argv, stdout=subprocess.PIPE,
+                            stderr=subprocess.STDOUT)
+    lines = result.stdout.decode("utf-8").split("\n")
+    if lines_containing:
+        lines = [line for line in lines if lines_containing in line]
+    return [cleanup_funny_output(line.strip()) for line in lines if line.strip()], result.returncode
+
+
+def answered_yes_to_confirmation(message: str) -> bool:
+    """
+    Prompts the user with the given message and asks for a yes or no answer,
+    and if yes is the user response then returns True, otherwise returns False.
+    """
+    response = input(f"{message} [yes | no]: ").lower()
+    if response == "yes":
+        return True
+    return False
+
+
+def exit_with_no_action() -> None:
+    """
+    Exits this process immediately with status 1;
+    first prints a message saying no action was taken.
+    """
+    PRINT("Exiting without taking action.")
+    exit(1)
```

Comment:
This is used in other places no? May belong in a file in the actual library...

Reply:
Maybe irrelevant point now, but before I realized (via Kent) that this could be shared, i.e. no need to be copied to each repo like the old publish bash script, my idea was that it should have absolute minimal dependencies, not even on dcicutils. But now that it is in dcicutils, I guess I could use dcicutils stuff, and including such a function as this that could go into misc_utils.